### PR TITLE
Add test to compare local hashes with a list of servers

### PR DESCRIPTION
**Pull request by ouziel-slama**
- this test is executed only with the flag `--comparehashes`
- this test needs a running local mainnet server
- will be executed after `dredd` in the Docker Compose test

### CODE REVIEW

**Review by adamkrellenstein — CHANGES_REQUESTED** [review]
This looks great, except I think we should test *all* supported versions and not just v9 vs v10. (That is, I think we should test v9.61.3, v10.00, v10.0.1, etc.)

**Comment by ouziel-slama** [comment]
> This looks great, except I think we should test _all_ supported versions and not just v9 vs v10. (That is, I think we should test v9.61.3, v10.00, v10.0.1, etc.)

yep that's mean to setup an instance for each version.. it's possible right now.. just need to add a list of servers in `CHECK_SERVERS` variable. `v9` means here to use the API v1 and `v10` means to use the API v2. I will fix that.

File: counterparty-core/counterpartycore/test/compare_hashes_test.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+from counterpartycore.test import (
+    conftest,  # noqa: F401
+)
+
+LOCAL_API_URL = "http://api:api@localhost:4000"
+
+# [server_url, api_version]
+CHECK_SERVERS = [
+    ["http://rpc:rpc@api1.counterparty.io:4000", "v1"],  # v9.61.3
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/test/compare_hashes_test.py:14]: hmm I think we should record which version is supposed to be running on each server and have this test verify that before passing

**Review by adamkrellenstein — APPROVED** [review]

File: counterparty-core/counterpartycore/test/compare_hashes_test.py (hunk 1 above)
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/test/compare_hashes_test.py]: yep good idea! done!
(so api1 is running v9.61.1)

**Pull request merged**